### PR TITLE
Claims and Role mappings documentation

### PR DESCRIPTION
**Pull request by enrichman**
Ref:
- https://github.com/epinio/epinio/pull/1816
- https://github.com/epinio/epinio/pull/1865
---

Documentation regarding the role mappings from the OIDC groups.

### CODE REVIEW

File: docs/references/authentication_oidc.md [hunk 1]
```diff
@@ -45,3 +45,32 @@ secret/ruser-adminepinioio-9341763ee7dcbce070e7c14f246ec8291e9a7278 patched
 
 You're now able to login with the credentials in Epinio UI.
 
+## Groups and Roles mapping
+
+The external identity providers can sometimes provide additional information about the user, for example the groups that he's member of.
```
Comment (andreas-kupries) [inline docs/references/authentication_oidc.md:50]:
```suggestion
The external identity providers may provide additional information about the user, for example the groups that he's member of.  
```

File: docs/references/authentication_oidc.md [hunk 2]
```diff
@@ -45,3 +45,32 @@ secret/ruser-adminepinioio-9341763ee7dcbce070e7c14f246ec8291e9a7278 patched
...
+```
+rolesMapping: |-
+  - connectorId: github
+    groups:
+    - id: Org1:Admins
+      role: admin
+    - id: Org1:TeamBlue
+      role: user
+
+config.yaml: |-
```
Comment (andreas-kupries) [inline docs/references/authentication_oidc.md:63]: What is the purpose of the `config.yaml` key ? It is not described in the preceding text, nor after.

File: docs/references/authentication_oidc.md [hunk 3]
```diff
@@ -45,3 +45,32 @@ secret/ruser-adminepinioio-9341763ee7dcbce070e7c14f246ec8291e9a7278 patched
...
+  - type: github
+    id: github
+    name: GitHub
+    config:
+      loadAllGroups: true
+      orgs:
+      - name: Org1
+```
+
+The groups will be evaluated in order. In the previous example if the user is a member of both the `Org1:Admins` and `Org1:TeamBlue` then he will get the `admin` role.
```
Comment (andreas-kupries) [inline docs/references/authentication_oidc.md:74]:
```suggestion
The groups will be evaluated in order. In the previous example if the user is a member of both `Org1:Admins` and `Org1:TeamBlue` then that user will get the `admin` role.
```

File: docs/references/authentication_oidc.md [hunk 4]
```diff
@@ -45,3 +45,32 @@ secret/ruser-adminepinioio-9341763ee7dcbce070e7c14f246ec8291e9a7278 patched
...
+
+Please note that the role is bound when the user is created, and any modification of the groups will not reflect to already existing users.
```
Comment (andreas-kupries) [inline docs/references/authentication_oidc.md:76]:
```suggestion
Please note that the role is bound when the user is created. Any modification of the groups will not propagate to already existing users.
```

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: docs/references/authentication_oidc.md (hunk 2 above)
Comment (enrichman) [reply docs/references/authentication_oidc.md:63]: It's the Dex configuration itself. At the beginning it was the only key in this secret, so it was straightforward. The we started adding more keys related on how Epinio integrates with Dex. Should I add something at the beginning maybe?

Line 23: https://github.com/epinio/docs/pull/191/files#diff-88a112af3f97d3ebffec4f38eede241145a6b74d7495746edec9caf36beae58dR23

Comment (andreas-kupries) [reply docs/references/authentication_oidc.md:63]: While the referenced line, i.e.

> To add more connectors, you can edit the Dex config file deployed with the `dex-config` secret

mentions the config file it is not fully clear that this "file" is the `config.yaml` key in the secret.
I would make that link a bit clearer.

Comment (andreas-kupries) [reply docs/references/authentication_oidc.md:63]: And when I said "preceding text" I only had looked at the text in the diff, not the entire file. That was my fault.

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**